R/timeseriessummary-data.R

```diff
@@ -71,6 +78,100 @@ formatDataTable <- function(inputData){
   return(returnData)
 }
 
+constructTSDetails <- function(reportData, timezone){
+  tsAttrs <- data.frame(stringsAsFactors = FALSE)
+  tsExtAttrs <- data.frame(stringsAsFactors = FALSE)
+  
+  
+  metadata <- parseTSSPrimaryTsMetadata(reportData)
+  methodData <- parseTSSMethods(reportData, timezone)
+  itData <- parseTSSInterpolationTypes(reportData, timezone)
+  processorData <- parseTSSProcessors(reportData, timezone)
+  changeNote <- FALSE
+  
+  if(!is.null(metadata) && !isEmptyOrBlank(metadata)){
+    #Time Series Attributes
+    tsAttrs <- rbind(tsAttrs, data.frame(label="Label", value=metadata[['identifier']], indent=8, stringsAsFactors = FALSE))
+    tsAttrs <- rbind(tsAttrs, data.frame(label="Parameter", value=metadata[['parameter']], indent=8, stringsAsFactors = FALSE))
+    tsAttrs <- rbind(tsAttrs, data.frame(label="Units", value=metadata[['unit']], indent=8, stringsAsFactors = FALSE))
+    if(!is.null(itData) && !isEmptyOrBlank(itData) && nrow(itData) > 0){
+      mIt <- FALSE
+      if(nrow(itData) > 1){
+        changeNote <- TRUE
+        mIt <- TRUE
+      }
+      tsAttrs <- rbind(tsAttrs, data.frame(label="Interpolation", value=ifelse(mIt,  paste(itData[1,][['type']], "*"), itData[1,][['type']]), indent=8, stringsAsFactors = FALSE))
+    }
+    tsAttrs <- rbind(tsAttrs, data.frame(label="Sub-Location", value=metadata[['sublocation']], indent=8, stringsAsFactors = FALSE))
+    tsAttrs <- rbind(tsAttrs, data.frame(label="UTC Offset", value=metadata[['utcOffset']], indent=8, stringsAsFactors = FALSE))
+    tsAttrs <- rbind(tsAttrs, data.frame(label="Computation", value=metadata[['computation']], indent=8, stringsAsFactors = FALSE))
+    tsAttrs <- rbind(tsAttrs, data.frame(label="Period", value=metadata[['period']], indent=8, stringsAsFactors = FALSE))
+    tsAttrs <- rbind(tsAttrs, data.frame(label="Publish", value=metadata[['publish']], indent=8, stringsAsFactors = FALSE))
+    tsAttrs <- rbind(tsAttrs, data.frame(label="Description", value=metadata[['description']], indent=8, stringsAsFactors = FALSE))
+    tsAttrs <- rbind(tsAttrs, data.frame(label="Comments", value=metadata[['comment']], indent=8, stringsAsFactors = FALSE))
+    if(!is.null(methodData) && !isEmptyOrBlank(methodData) && nrow(methodData) > 0){
```

Comment:
`isEmptyOrBlank` does `is.null` already; use `isEmptyVar` instead of `nrow() > 0`